src/MobileRedirection.php

```diff
@@ -49,6 +49,10 @@ public function register() {
 
 		if ( AMP_Options_Manager::get_option( Option::MOBILE_REDIRECT ) ) {
 			add_action( 'template_redirect', [ $this, 'redirect' ], PHP_INT_MAX );
+
+			// Enable AMP-to-AMP linking by default to avoid redirecting to AMP version when navigating.
+			// A low priority is used so that sites can continue overriding this if they have done so.
+			add_filter( 'amp_to_amp_linking_enabled', '__return_true', 0 );
```

Comment:
I'm probably missing something here, but why is the `'amp_to_amp_linking_enabled'` filter even at all coupled to the state of `Option::MOBILE_REDIRECT` ? Wouldn't the filter be something you set for your AMP presence regardless of whether you redirect on mobile?

Reply:
Because if you have mobile redirection enabled, you don't want mobile visitors navigating around AMP pages to be redirected for every navigation. By adding AMP-to-AMP linking, no further redirections will happen once on an AMP page.

Reply:
The priority of `0` is to not conflict with sites that are adding this filter already to manually control the behavior.

Reply:
This is also [mentioned](https://github.com/ampproject/amp-wp/releases/tag/2.0.0#~:text=When%20mobile%20redirection%20is%20enabled%2C%20AMP%2Dto%2DAMP%20linking%20is%20automatically%20enabled.) in the release notes:

![image](https://user-images.githubusercontent.com/134745/92153212-c1e5c080-edd8-11ea-8f71-f55731bcfec6.png)